### PR TITLE
Bucketlist endpoints

### PR DESCRIPTION
**Pull request by NgangaMaryanne**
#### What does this PR do?
Add bucketlist endpoints to the API

#### Description of Task to be completed?
A logged in user should be able to view, update, delete and post bucketlists and bucketlist items.
#### How should this be manually tested?
* Clone repository
* Test the endpoints on postman

### CODE REVIEW

File: bucketlist/api_v1/views.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        return f(*args, **kwargs)
+    return decorated_function
+
+    
+
+class BucketlistApi(Resource):
+    '''
+    Bucketlist class for all bucketlists
+    '''
+    # decorators=[auth.login_required]
```
Comment (gacheruevans) [inline bucketlist/api_v1/views.py]: remove commented code

File: bucketlist/api_v1/views.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+                            'message':str(Exception)}
+                return make_response(jsonify(response))
+        else:
+            response = {'status':'fail',
+                        'message':'Bucketlist with that name already exists. please try again.'
+                        }
+            make_response(jsonify(response))
+    @authentication_required
+    def put(self, bucket_id):
+        parser = reqparse.RequestParser()
```
Comment (andela-gacheruevans) [inline bucketlist/api_v1/views.py]: use doc strings to give a brief description of the function's role

File: bucketlist/api_v1/views.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+                            'message':str(SQLAlchemyError)}
+                return make_response(jsonify(response))
+
+        else:
+            response = {'message':'You do not have a bucket with id {0}'.format(bucket_id)}
+            return make_response(jsonify(response))
+
+    @authentication_required
+    def delete(self, bucket_id):
+        bucket = Bucketlist.query.filter_by(id=bucket_id, created_by=int(g.user)).first()
```
Comment (andela-gacheruevans) [inline bucketlist/api_v1/views.py]: use doc strings to give a brief description of the function's role

File: bucketlist/api_v1/views.py (hunk 1 above)
Comment (NgangaMaryanne) [reply bucketlist/api_v1/views.py]: done

**Pull request merged**